onnxruntime/core/optimizer/graph_transformer_utils.cc

```diff
@@ -177,6 +181,10 @@ InlinedVector<std::unique_ptr<GraphTransformer>> GenerateTransformers(
 
       // no filtering on execution provider for L1 optimizations as they only use official ONNX operators
       transformers.emplace_back(std::make_unique<CommonSubexpressionElimination>());
+#ifdef USE_XNNPACK
+      auto default_cpu_allocator = cpu_execution_provider.GetAllocator(0, OrtMemTypeDefault);
+      transformers.emplace_back(std::make_unique<XNNPackTransformer>(default_cpu_allocator));
+#endif
```

Comment:
It should not be a level 1 optimization. Or it prevent other EPs

Reply:
It has ifdef, so it won't impact the other builds. I put it here because the transformer highly depends on constant folding. Constant folding must be invoked if the graph was changed by this transformer. So I have to put it before constant folding. There are some other possible solutions:
1. Manually transpose weights. Then the downside is transpose optimizer would not take effect too. Also I need to refactor the current Tranpose kernel so that I can reuse its code in graph transformers.
2.  Manually invoke constant folding in this transformer. Normally we don't do such things. If one transformer calls another, it's easy to construct a dead loop.

Reply:
call MlasTranspose: [https://github.com/microsoft/onnxruntime/blob/aaa583e776ef2dce8dc4748f1303a092e5b2e634/onnxruntime/core/providers/cpu/quantization/qlinearconv.cc#L887](https://github.com/microsoft/onnxruntime/blob/aaa583e776ef2dce8dc4748f1303a092e5b2e634/onnxruntime/core/providers/cpu/quantization/qlinearconv.cc#L887)?